app/models/user_field.rb

```diff
@@ -14,9 +14,12 @@ class UserField < ActiveRecord::Base
   accepts_nested_attributes_for :user_field_options
 
   before_save :sanitize_description
+  after_create :update_required_fields_version
+  after_update :update_required_fields_version, if: -> { saved_change_to_requirement? }
   after_save :queue_index_search
 
   scope :public_fields, -> { where(show_on_profile: true).or(where(show_on_user_card: true)) }
+  scope :required, -> { where.not(requirement: 0) }
```

Comment:
Here we can just use the scope automatically created by the `enum`: `not_optional` :slightly_smiling_face:

Reply:
I forgot this was added to Rails. Thank you, @Flink! ☺️